tests/0081-admin.c

```diff
@@ -3652,239 +3652,335 @@ static void do_test_ListConsumerGroupOffsets(const char *what,
-                        /* username should be broker , errorcode should be 91 and num_credentials should be 0 */
-                        TEST_ASSERT(strcmp(users[0],username)==0,"Username does not match !");
-                        TEST_ASSERT(((errorcode == 91) && (num_credentials == 0)),"Error-Code should be 91 as user does not exist and credentials count should be 0");
-                        
-                }
+
+        TEST_CALL_ERR__(rd_kafka_event_error(event));
+        err = rd_kafka_event_error(event);
+        TEST_ASSERT(err == RD_KAFKA_RESP_ERR_NO_ERROR,
+                    "Expected NO_ERROR, not %s", rd_kafka_err2name(err));
+
+        describe_result =
+            rd_kafka_event_DescribeUserScramCredentials_result(event);
+        descriptions =
+            rd_kafka_DescribeUserScramCredentials_result_descriptions(
+                describe_result, &description_cnt);
+        /* Assert num_results should be 1 , request level error code should be
+         * 0*/
+        TEST_ASSERT(description_cnt == 1,
+                    "There should be exactly 1 description, got %" PRIusz,
+                    description_cnt);
+        for (i = 0; i < description_cnt; i++) {
+                const rd_kafka_UserScramCredentialsDescription_t *description =
+                    descriptions[i];
+                const char *username;
+                const rd_kafka_error_t *error;
+                username =
+                    rd_kafka_UserScramCredentialsDescription_user(description);
+                error =
+                    rd_kafka_UserScramCredentialsDescription_error(description);
+                rd_kafka_resp_err_t err = rd_kafka_error_code(error);
+
+                size_t num_credentials =
+                    rd_kafka_UserScramCredentialsDescription_scramcredentialinfo_count(
+                        description);
+                /* username should be the same, err should be RESOURCE_NOT_FOUND
+                 * and num_credentials should be 0 */
+                TEST_ASSERT(strcmp(users[0], username) == 0,
+                            "Username should be %s, got %s", users[0],
+                            username);
+                TEST_ASSERT(err == RD_KAFKA_RESP_ERR_RESOURCE_NOT_FOUND,
+                            "Error code should be RESOURCE_NOT_FOUND as user "
+                            "does not exist, got %s",
+                            rd_kafka_err2name(err));
+                TEST_ASSERT(num_credentials == 0,
+                            "Credentials count should be 0, got %" PRIusz,
+                            num_credentials);
         }
         rd_kafka_event_destroy(event);
 
+        /* Create a credential for user 0 */
+
         rd_kafka_UserScramCredentialAlteration_t *alterations[1];
-        char *salt = "salt";
-        char *password = "password";
+        char *salt                          = "salt";
```

Comment:
we should change the testcase to deliberately have a zero-byte between the salt, that way, we get an extra check on the serialization to bytes as well.